src/modules/peek/Peek.FilePreviewer/Previewers/ShellPreviewHandlerPreviewer/ShellPreviewHandlerPreviewer.cs

```diff
@@ -0,0 +1,185 @@
+using Windows.Win32.UI.Shell;
+using Windows.Win32.UI.Shell.PropertiesSystem;
+
+namespace Peek.FilePreviewer.Previewers
+{
+    public partial class ShellPreviewHandlerPreviewer : ObservableObject, IShellPreviewHandlerPreviewer, IDisposable
+    {
+        [ObservableProperty]
+        private IPreviewHandler? preview;
+
+        [ObservableProperty]
+        private PreviewState state;
+
+        private Stream? fileStream;
+
+        public ShellPreviewHandlerPreviewer(IFileSystemItem file)
+        {
+            FileItem = file;
+            Dispatcher = DispatcherQueue.GetForCurrentThread();
+        }
+
+        private IFileSystemItem FileItem { get; }
+
+        private DispatcherQueue Dispatcher { get; }
+
+        public void Dispose()
+        {
+            Clear();
+            GC.SuppressFinalize(this);
+        }
+
+        public async Task CopyAsync()
+        {
+            await Dispatcher.RunOnUiThread(async () =>
+            {
+                var storageItem = await FileItem.GetStorageItemAsync();
+                ClipboardHelper.SaveToClipboard(storageItem);
+            });
+        }
+
+        public Task<PreviewSize> GetPreviewSizeAsync(CancellationToken cancellationToken)
+        {
+            return Task.FromResult(new PreviewSize { MonitorSize = null });
+        }
+
+        public async Task LoadPreviewAsync(CancellationToken cancellationToken)
+        {
+            Clear();
+            State = PreviewState.Loading;
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Create the preview handler
+            var previewHandler = await Task.Run(() =>
+            {
+                var previewHandlerGuid = GetPreviewHandlerGuid(FileItem.Extension);
+                if (!string.IsNullOrEmpty(previewHandlerGuid))
+                {
+                    return Activator.CreateInstance(Type.GetTypeFromCLSID(Guid.Parse(previewHandlerGuid))!) as IPreviewHandler;
+                }
+                else
+                {
+                    return null;
+                }
+            });
+
+            if (previewHandler == null)
+            {
+                State = PreviewState.Error;
+                return;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Initialize the preview handler with the selected file
+            bool success = await Task.Run(() =>
+            {
+                const uint STGM_READ = 0x00000000;
+                if (previewHandler is IInitializeWithStream initWithStream)
+                {
+                    fileStream = File.OpenRead(FileItem.Path);
+                    initWithStream.Initialize(new IStreamWrapper(fileStream), STGM_READ);
+                }
+                else if (previewHandler is IInitializeWithFile initWithFile)
+                {
+                    unsafe
+                    {
+                        fixed (char* pPath = FileItem.Path)
```

Comment:
I am not extremely familiar with a fixed context, but is it really needed here? Could the garbage collector mess with something in here?

Reply:
Same with above.